include/zephyr/net/net_event.h

```diff
@@ -194,24 +194,29 @@ enum net_event_ipv4_cmd {
 				 NET_MGMT_LAYER_CODE(_NET_L4_CORE_CODE))
 
 enum net_event_l4_cmd {
-	NET_EVENT_L4_CMD_CONNECTED = 1,
```

Comment:
Just wondering why we have "IF" (meaning network interface) here as at L4 there should be no concept of network interface. L3 has network interface because we have IP address discovery at that layer, but in L4 we should be dealing with connectivity issues in the whole system, not just something that concerns one interface.

Reply:
Yes, I agree that it is odd that the event is part of the L4 event group, but this commit aims only to rename the event to have a more accurate name, rather than change its behavior or which event group it lives in.

Regardless of whether it is named `NET_EVENT_L4_CMD_CONNECTED` or `NET_EVENT_L4_IF_CMD_READY`, it is being called per-iface by conn_mgr.

Changing the group it lives in is a much more invasive change, and probably beyond the scope of this PR